target/linux/mpc85xx/image/p1010.mk

```diff
@@ -14,6 +14,26 @@ define Build/spi-loader-okli
 	mv "$@.new" "$@"
 endef
 
+define Device/aerohive_br200-wp
+  DEVICE_VENDOR := Aerohive
+  DEVICE_MODEL := BR200-WP
+  BLOCKSIZE := 128k
+  KERNEL_NAME := simpleImage.br200-wp
+  KERNEL := kernel-bin | uImage none
+  KERNEL_INITRAMFS := kernel-bin | uImage none
+  KERNEL_ENTRY := 0x1000000
+  KERNEL_LOADADDR := 0x1000000
+  KERNEL_SIZE := 8m
+  IMAGES := fdt.bin sysupgrade.bin
+  IMAGE/fdt.bin := append-dtb
+  IMAGE/sysupgrade.bin := append-dtb | pad-to 256k | check-size 256k | \
```

Comment:
Seems like, as with other mpc85xx Aerohives, you've found a hardcoded bootcmd; with an 8m limit on kernel-size, and using an uncompressed kernel, won't we be running out of space fairly soon?

Reply:
I use simpleImage with LZMA self-extractor:
https://docs.kernel.org/powerpc/bootwrapper.html

Now kernel have 3MB only:
```

U-Boot 2009.11 (Sep 17 2015 - 00:35:03)

CPU:   P1011E, Version: 1.1, (0x80ed0011)
Core:  E500, Version: 5.1, (0x80212051)
Clock Configuration:
       CPU0:660  MHz,
       CCB:330  MHz,
       DDR:330  MHz (660 MT/s data rate) (Asynchronous), LBC:20.625 MHz
L1:    D-cache 32 kB enabled
       I-cache 32 kB enabled
I2C:   ready
SPI:   ready
DRAM:  Configuring DDR for 660 MT/s data rate
DDR: 256 MB
FLASH: 64 MB
L2:    256 KB enabled
MMC:

    PCIE2: disabled

    PCIE1 connected to Slot 2 as Root Complex (base addr ffe0a000)
               Scanning PCI bus 01
        01  00  168c  0030  0280  ff
    PCIE1 on bus 00 - 01

In:    serial
Out:   serial
Err:   serial
Net:   initialization for Atheros AR8327/AR8328,addr 4
AR8327/AR8328 v1.1 is found!
set for PHY4
initialization for Atheros AR8327/AR8328,addr 0
eth0, eth1
Hit any key to stop autoboot:  0
Hit any key to interrupt boot from flash:  0
## Booting kernel from Legacy Image at ee840000 ...
   Image Name:   POWERPC OpenWrt Linux-5.15.106
   Created:      2023-04-12  11:21:21 UTC
   Image Type:   PowerPC Linux Kernel Image (uncompressed)
   Data Size:    3094436 Bytes =  3 MB
   Load Address: 01000000
   Entry Point:  01000000
   Verifying Checksum ... OK
## Loading init Ramdisk from Legacy Image at 02000000 ...
   Image Name:   OpenWrt fake ramdisk
   Created:      2023-04-12  11:21:21 UTC
   Image Type:   PowerPC Linux RAMDisk Image (uncompressed)
   Data Size:    0 Bytes =  0 kB
   Load Address: 00000000
   Entry Point:  00000000
   Verifying Checksum ... OK
## Flattened Device Tree blob at ec000000
   Booting using the fdt blob at 0xec000000
   Loading Kernel Image ... OK
OK
   Loading Ramdisk to 10000000, end 10000000 ... OK
   Loading Device Tree to 00ff9000, end 00fff178 ... OK
ft_fixup_l2cache: FDT_ERR_NOTFOUND

zImage starting: loaded at 0x01000000 (sp: 0x012f4fa0)
Allocating 0x91626c bytes for kernel...
Decompressing (0x00000000 <- 0x01014000:0x012f37a4)...
Done! Decompressed 0x8dd7fc bytes

Linux/PowerPC load: console=ttyS0,9600
Finalizing device tree... flat tree at 0x12f5cc0
[    0.000000] Memory CAM mapping: 256 Mb, residual: 0Mb
[    0.000000] Linux version 5.15.106 (krolik@ubuntu) (powerpc-openwrt-linux-musl-gcc (OpenWrt GCC 12.2.0 r21408+1090-7dbac3433fef) 12.2.0, GNU ld (GNU Binutils) 2.40.0) #0 Wed Apr 12 11:21:21 2023
```

Reply:
GAH! How did we not notice this when we were fighting with the [Aerohive HiveAP 330](https://github.com/openwrt/openwrt/pull/4897)?

It must have just been who was around at the time. simpleImage comes with its own challenges, clearly they've been figured out for the host of mpc85xx boards OpenWrt supports.